### PR TITLE
Updage to Angular 14, TypeScript 4.6 and RxJS 7.5

### PR DESCRIPTION
**Pull request by fjaguado**
# PR Details

Update to Angular 14 and TypeScript 4.6
Fix some builder issues while updating RxJS to 7.5

## Description

Followed Angular Update Guide

The update to version 13 has been done first
https://update.angular.io/?l=3&v=12.0-13.0 but adding @angular-eslint/schematics@13 to the ng update

And then to version 14 following the same pattern
https://update.angular.io/?l=3&v=13.0-14.0 but adding @angular-eslint/schematics@14 to the ng update

At this point the following libraries have been updated or the version are fixed: 

- "eslint-plugin-import": "2.26.0"
- "eslint-plugin-jsdoc": "39.3.6"
- "eslint-plugin-prefer-arrow": "1.2.3"
- "karma-jasmine-html-reporter": "1.5.4"
- "pako":"^2.0.4"

The following packages have been removed because they are deprecated:

- "@angular-devkit/build-ng-packagr": "^0.1002.0" 

Instead of build-ng-packagr we moved to build-angular and changed a line in angular.json

- "builder": "@angular-devkit/build-angular:ng-packagr"

Also the Travis configuration has been updated

## Related Issue
#56 

## Motivation and Context

This update is part of a cybersecurity demand on how update a full system can be achieved. Also, we want to have the latest technology too.

## How has this been tested

It has been executed tests

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the CONTRIBUTING document
- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A new branch needs to be created from master to evolve previous versions
- [x] Increase version in package.json following [Semantic Versioning](https://semver.org/)
- [x] Add the issue into the right [project](https://github.com/systelab/systelab-translate/projects) with the proper status (In progress)

### CODE REVIEW

**Comment by OscarDeDios** [comment]
Add to the readme a comment about the new major version and the angular upgrade please

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+  - '14.20.1'
 addons:
   chrome: stable
 
+services:
+  - xvfb
+
 before_install:
-  - export CHROME_BIN=chromium-browser
+  - export CHROME_BIN=/usr/bin/google-chrome
```
Comment (carlosra85) [inline .travis.yml:13]: why is this needed?

Comment (fjaguado) [reply .travis.yml:13]: Because of new versions of Ubuntu used in Travis

File: projects/systelab-preferences/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "systelab-preferences",
-  "version": "9.1.0",
+  "version": "14.0.0",
```
Comment (carlosra85) [inline projects/systelab-preferences/package.json:3]: I like the idea to adapt versioning to Angular versions, maybe we need to discuss this in next task force meeting to start doing the same in all applications.

**Review by carlosra85 — APPROVED** [review]

File: projects/systelab-preferences/package.json (hunk 2 above)
Comment (fjaguado) [reply projects/systelab-preferences/package.json:3]: Probably is a good idea for production applications. The problem probably is that the version of the company application not only are dependent from Angular. For example, in Beacon we are using Java and Sprinboot

Comment (carlosra85) [reply projects/systelab-preferences/package.json:3]: I meant in Systelab libraries, to specify clearly in the versioning semantics that major version will match from now on Angular version

**Review by OscarDeDios — APPROVED** [review]

**Review by OscarDeDios — APPROVED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -1,21 +1,23 @@
 sudo: required
-dist: trusty
+dist: jammy
 language: node_js
 node_js:
-  - '12'
+  - '14.20.1'
```
Comment (joaquimvila) [inline .travis.yml:5]: Same comment as in systelab-translate: shouldn't we move the version 16 as this is the latest LTS?
According to the docs, the 16.13.0 is preinstalled on the "jammy" image: https://docs.travis-ci.com/user/reference/jammy/#javascript-and-nodejs-support

**Review by joaquimvila — APPROVED** [review]
Good job @fjaguado

**Review by RubKrmon — APPROVED** [review]

File: .nvmrc [hunk 4]
```diff
@@ -0,0 +1 @@
+v16.17.1
```
[the remark this answers is not in the document]
Comment (fjaguado) [reply .nvmrc:1]: Yes, it's useful but not in Windows. I forgot to remove it, I'll do it. It has no sense to have this file if the full company is working on Windows
